tests/post/steps/test_monitoring.py

```diff
@@ -81,6 +97,31 @@ def _check_object_exists():
     utils.retry(_check_object_exists, times=20, wait=3)
 
 
+# FIXME: make these fixtures / helpers generic and easier to share
+@given("I have created a test Volume")
+def test_volume(k8s_apiclient, ssh_config):
+    client = kube_utils.VolumeClient(
+        CustomObjectsApi(api_client=k8s_apiclient), ssh_config
+    )
+    random_salt = ''.join(
+        random.choice(string.ascii_lowercase) for _ in range(8)
+    )
+    volume_name = "test-volume-{}".format(random_salt)
+    client.create_from_yaml(DEFAULT_VOLUME.format(name=volume_name))
+
+    def _check_volume_ready():
+        volume = _get_volume_and_check_status(client, volume_name, 'Available')
+        assert volume['status'].get('deviceName') is not None, \
+            'Volume status.deviceName has not been reconciled'
+        return volume
+
+    yield utils.retry(
+        _check_volume_ready, times=30, wait=2,
+        name='waiting for Volume {} to be ready'.format(volume_name)
+    )
```

Comment:
Can't reuse `check_volume_status` for that one?

Reply:
They are slightly different actually:
- this one checks if there's a device name - so it's a bit more than just checking the status phase
- `check_volume_status` is used in `test_volume` scenarii to manipulate the `context` fixture - we don't need it here

Reply:
Ok fair enough.